internal/proxy/task_scheduler.go

```diff
@@ -228,16 +228,22 @@ type dmTaskQueue struct {
 func (queue *dmTaskQueue) Enqueue(t task) error {
 	queue.statsLock.Lock()
 	defer queue.statsLock.Unlock()
-	err := queue.addPChanStats(t)
+	//1. preAdd will check whether provided task is valid or addable
+	//and get the current pChannels for this dmTask
+	pChannels, dmt, err := queue.preAddPChanStats(t)
```

Comment:
preAddPChanStats just generate channels and dmlTask?

Reply:
to check whether all channels are available and the input task can be reflected to dmlTask
if these two conditions are not fulfilled, preAdd will fail and ensure the atomic property of the whole process